src/demo.rs

```diff
@@ -0,0 +1,111 @@
+    loop {
+        tokio::select! {
+            // Handle file events
+            Some(event) = rx.recv() => {
+                match event.kind {
+                    notify::event::EventKind::Create(_) => {
+                        let path = event.paths[0].clone();
+                        if path.extension().map_or(false, |ext| ext == "dem") {
+                            current_file_path = path.to_string_lossy().to_string();
+                            current_file_position = 0;
+                            tracing::info!("New .dem file created: {}", &current_file_path);
+                        } else {
+                            tracing::info!("Ignored file with non-.dem extension: {}", path.to_string_lossy());
+                        }
+                    }
+                    notify::event::EventKind::Modify(_) => {
+                        let path = event.paths[0].clone();
+                        if path.to_string_lossy() == current_file_path {
+                            let mut file = File::open(&current_file_path).await?;
+                            let mut buffer = Vec::new();
+
+                            file.seek(SeekFrom::Start(current_file_position)).await?;
+                            let read_bytes = file.read_to_end(&mut buffer).await?;
+                            current_file_position += read_bytes as u64;
+
+                            if buffer.len() != 0{
+                                process_file_data(buffer).await;
+                            }
+                        }
+                    }
+                    _ => {
+                        tracing::error!("Unhandled event kind: {:?}", event.kind);
+                    }
+                }
+            },
+            // Handle metadata tick
+            _ = metadata_tick.tick() => {
+                // If there is a current file being watched, check its metadata
+                if !current_file_path.is_empty() {
+                    let current_metadata = metadata(&current_file_path).await?;
+                    let current_size = current_metadata.len();
+
+                    // If the current file size is greater than the last known position, it's been updated
+                    if current_size > current_file_position {
+                        let mut file = File::open(Path::new(&current_file_path)).await?;
+                        let mut buffer = Vec::new();
+                        file.seek(SeekFrom::Start(current_file_position)).await?;
+                        let read_bytes = file.read_to_end(&mut buffer).await?;
+                        current_file_position += read_bytes as u64;
+                        if buffer.len() != 0{
+                            process_file_data(buffer).await;
+                        }
+                    }
+                }
+            },
+        }
+    }
+}
+
+async fn process_file_data(data: Vec<u8>) {
+    // Placeholder function
+    tracing::info!("Received data of length {}", data.len());
```

Comment:
Some of these `tracing::info` could probably be `tracing::debug` for now to keep the console a bit cleaner for most users.